lib/elixir/src/elixir_expand.erl

```diff
@@ -1165,6 +1181,12 @@ format_error({struct_comparison, StructExpr}) ->
                 "Comparing with a struct literal is unlikely to give a meaningful result. "
                 "Modules typically define a compare/2 function that can be used for "
                 "semantic comparison", [String]);
+format_error({nested_comparison, CompExpr}) ->
+  String = 'Elixir.Macro':to_string(CompExpr),
+  io_lib:format("nested comparisons are discouraged. Instead of ~ts,"
+                "consider joining together each comparison segment "
+                "with an &&.", [String]);
```

Comment:
My take would be this:

```erlang
"Elixir does not supported nested comparisons. Something like\n\n"
"    x < y < z\n\n"
"is equivalent to:\n\n"
"    (x < y) < z\n\n"
"which would ultimately compare the result of (x < y), which is a boolean, with z. "
"Instead consider joining together each comparison segment with an \"and\", such as\n\n"
"    x < y and y < z\n\n"
"You wrote: ~ts"
```

Reply:
Makes sense -- I've included this with one minor tweak